replaced the print statements with logging (issue #59)

ChristensenCode opened the pull request:
Did a find all for print statements in the code and replaced them with python's logging module instead and added a constants file to store the common logging format.
This is related to issue #59

ChristensenCode committed:
gitignore should ignore all pyc files, not just those specific to python 37

ChristensenCode committed:
added a setup to store common objects and teardown to clean up created files

ChristensenCode committed:
added some setup and teardown and removed unneeded import

ChristensenCode committed:
started refactoring the units class

ChristensenCode committed:
started adding tests for unit class

ChristensenCode committed:
updated test class and added unit tests

ChristensenCode committed:
updated the constants and placed them in their own area for easier reference.

ChristensenCode committed:
started using the constants and unit conversion in chiller

ChristensenCode committed:
started documenting the origin of the constants and left the origin units. it'd be too much to change all the unit calls right now.

ChristensenCode committed:
minor readme change and made gitignore my local vscode stuff

ChristensenCode committed:
brought back the original units test

ChristensenCode committed:
fixed minor type

ChristensenCode committed:
started adding information about the input type

ChristensenCode committed:
filled in all i could about the type of input the CLI expects.

ChristensenCode committed:
useful for a first time user to have a starting point

ChristensenCode committed:
replaced the print statements with logging

ChristensenCode committed:
replaced the print statements with logging

ChristensenCode committed:
schema for checking json file

ChristensenCode committed:
cleaned up imports, but now tests are failing. i need to fix the eval stuff next by removing it

ChristensenCode committed:
created json input reader

ChristensenCode committed:
Revert "schema for checking json file"

This reverts commit 5e24ef082e39e8135a368c663ca0a383a06f3ffd.

ChristensenCode committed:
Revert "cleaned up imports, but now tests are failing. i need to fix the eval stuff next by removing it"

This reverts commit a0627fb96c7a7ce9fddae08eb57bc34f9aef9e2e.

ChristensenCode committed:
added instruction set reader, but it still needs tests

ChristensenCode committed:
blacked the files and added some instruction set reader logic to the library file

ChristensenCode committed:
blacked the tests

lymereJ commented on copper/constants.py:
```diff
@@ -0,0 +1,37 @@
+# https://www.ahrinet.org/sites/default/files/2022-06/AHRI%20Standard%20551-591%202020%20%28SI%29%20with%20Addendum%201.pdf
+TOWER_ECT_551_591 = 35.00
+TOWER_ENTERING_TEMP_551_591 = 30.00
+COOLING_LCT_551_591 = 7.0
```
This is not in related to logging so it should probably not included in this PR.

lymereJ commented on copper/constants.py:
```diff
@@ -0,0 +1,37 @@
+COOLING_LCT_551_591 = 7.0
+
+# Format for the logger when it logs messages
+LOGGING_FORMAT = "%(filename)s:%(lineno)d:%(levelname)s:%(message)s"
```
I think that it might be easier to move all `logging` setup to the `__init__.py` file.

lymereJ replied:
Since most of the message are informative, I'm thinking that the format should probably be less file/code centric and be more general, perhaps something like `format='%(asctime)s - %(levelname)s - %(message)s'`.

lymereJ replied:
We should probably output everything to stdout, so we'll need `stream=sys.stdout` in `basicConfig()`

lymereJ commented on copper/curves.py:
```diff
@@ -11,9 +11,13 @@
 from copper.units import *
+from copper.constants import LOGGING_FORMAT
+
+logging.basicConfig(format=LOGGING_FORMAT)
```
If we put all the logger configuration in `__init__.py` then we can just retrieve it using `logger = logging.getLogger(__name__)`

lymereJ reviewed (changes requested):
@ChristensenCode - Sorry for the belated follow-up, I was out of the country for a few weeks. I've added a few comments. Please let me know if you think you can/want to address them this week, this is tied to a submission to the JOSS so I'd like to wrap it. I'll make the changes if you don't have time this week.

In addition to these comments, it would be nice to have a basic unit test for it.

ChristensenCode committed:
added basicconfig to init instead of in every file

ChristensenCode committed:
logging format no longer required

ChristensenCode committed:
removed constants since they are not part of logging change

ChristensenCode committed:
removed reference to logging format

ChristensenCode committed:
added test for new logging

ChristensenCode committed:
Merge branch 'develop' of https://github.com/pnnl/copper into prints_to_logging

ChristensenCode committed:
Merge branch 'develop' of https://github.com/ChristensenCode/copper into prints_to_logging

ChristensenCode committed:
reverted curve change since constant updated

ChristensenCode committed:
removing new units and constants reference that came in from other branch

ChristensenCode committed:
fixed logging test

ChristensenCode committed:
Revert "added test for new logging"

This reverts commit beb35e1fc49b9243e1cdcb44d745f25dfb842509.

ChristensenCode committed:
added test for logging

ChristensenCode committed:
Merge branch 'develop' of https://github.com/ChristensenCode/copper into prints_to_logging

ChristensenCode committed:
including document changes from develop

ChristensenCode committed:
fixed link in quickstart guide

ChristensenCode commented:
@lymereJ My apologies for all the extra commits. I got turned around on what branch I was on and kept going back and forth.
I changed the format of the logging string and moved the basicconfig to the __init__. I also added a test for logging. I didn't see anything great (other than mocks or additional libraries) for testing the literal string being printed to the screen.

lymereJ commented on .gitignore:
```diff
@@ -1,7 +1,8 @@
 /dev/
 /bin/
 /dist/
+**/.vscode/**
```
Could you please revert these changes?

ChristensenCode replied:
changes reverted

lymereJ commented on copper/__init__.py:
```diff
@@ -1 +1,5 @@
+from copper.constants import LOGGING_FORMAT
+import sys
+
+logging.basicConfig(format=LOGGING_FORMAT, stream=sys.stdout)
```
Could you please add `level=logging.INFO` at the end? That way we'll get everything displayed, we might want to change it later but for now it might be best.

ChristensenCode replied:
changed the logging level

lymereJ commented on copper/constants.py:
```diff
@@ -0,0 +1,7 @@
+constants.py
+====================================
+Holds all the constants referenced in copper.
+TODO: there are additional constants, but i haven't found them yet.
```
Maybe let's remove that TODO for now.

ChristensenCode replied:
removed todo

lymereJ reviewed (changes requested):
@ChristensenCode - No worries! The test looks good enough to me, thanks for adding it. I left a few minor comments, once addressed I'll merge the PR.

ChristensenCode committed:
addressing comments from @lymereJ

lymereJ commented:
Thanks a lot for this effort, great addition to the code!
